bundles/org.openhab.binding.teleinfo/src/main/java/org/openhab/binding/teleinfo/internal/reader/io/TeleinfoInputStream.java

```diff
@@ -120,20 +121,30 @@ public TeleinfoInputStream(final @Nullable InputStream teleinfoInputStream, long
             throw new IllegalArgumentException("Teleinfo inputStream is null");
         }
 
-        this.waitNextHeaderFrameTimeoutInMs = waitNextHeaderFrameTimeoutInMs;
+        this.waitNextHeaderFrameTimeoutInUs = waitNextHeaderFrameTimeoutInMs;
         this.readingFrameTimeoutInMs = readingFrameTimeoutInMs;
         this.autoRepairInvalidADPSgroupLine = autoRepairInvalidADPSgroupLine;
 
         this.bufferedReader = new BufferedReader(new InputStreamReader(teleinfoInputStream, StandardCharsets.US_ASCII));
+        this.executorService = Executors.newFixedThreadPool(2);
 
         groupLine = null;
     }
 
+    public TeleinfoInputStream(final @Nullable InputStream teleinfoInputStream, long waitNextHeaderFrameTimeoutInMs,
+            long readingFrameTimeoutInMs, boolean autoRepairInvalidADPSgroupLine, ExecutorService executorService) {
+        this(teleinfoInputStream, waitNextHeaderFrameTimeoutInMs, readingFrameTimeoutInMs,
```

Comment:
The invocation of this constructor will indirectly create the thread pool in line 129.